snuba/manual_jobs/__init__.py

```diff
@@ -43,12 +36,19 @@ def error(self, line: str) -> None:
 class JobSpec:
     job_id: str
     job_type: str
+    is_async: Optional[bool] = False
     params: Optional[MutableMapping[Any, Any]] = None
 
 
 class Job(ABC, metaclass=RegisteredClass):
     def __init__(self, job_spec: JobSpec) -> None:
         self.job_spec = job_spec
+        self.is_async = job_spec.is_async
+        _set_job_is_async(
+            job_spec.job_id,
+            job_spec.is_async if job_spec.is_async is not None else False,
+        )
```

Comment:
do we need to set a special field to track whether a job is async?

could we just detect that by looking at a job status and seeing if it's in `async_running_background`?

Reply:
you are right